niceman/distributions/base.py

```diff
@@ -194,7 +194,7 @@ def _init(self):
 
     @abc.abstractmethod
     def identify_distributions(self, files):
-        raise NotImplementedError()
+        return
```

Comment:
may be indeed pointless, but how plain `return` is better (if it is not better - why bother/make it worse)?
at least ` raise NotImplementedError` serves as some kind of an additional guard if `abc` and metaclass stuff doesn't work.

Reply:
> may be indeed pointless, 

It is.

> but how plain `return` is better (if it is not better - why bother/make it worse)?

`return` or `pass` is required to avoid a syntax error unless there's a docstring.  Depending on taste (consistency versus necessity), they could be dropped if there is a docstring.  But they are just simple stand-ins.  Doing anything more involved is likely to confuse a reader because (1) it's unconventional and (2) it suggests that the code is reachable because otherwise why bother.

> at least ` raise NotImplementedError` serves as some kind of an additional guard if `abc` and metaclass stuff doesn't work.

That instantiation with abstract methods should raise a `TypeError` is by design (PEP 3119).

Reply:
> That instantiation with abstract methods should raise a TypeError is by design (PEP 3119).

that was my point that "should" might not work, as e.g. you have discovered that `__metaclass__ = abc.ABCMeta` now might not be the right way to go.  With `pass` or `return` you would just miss that such protection stopped working for one reason or another.  Explicitly raised exception wouldn't.   But I am not going to battle it further since hopefully one way or another it wouldn't make difference in the future.